tests/functional/worker/notice-error.test.js

```diff
@@ -0,0 +1,118 @@
+ */
+
+const testDriver = require('../../../tools/jil/index')
+const {workerTypes, typeToMatcher} = require('./helpers')
+const {fail} = require('../err/assertion-helpers')  // shared from jserrors feat tests
+
+const init = {
+  jserrors: {
+    harvestTimeSeconds: 5
+  },
+  metrics: {
+    enabled: false
+  }
+}
+
+workerTypes.forEach(type => {
+  noticeErrorTest(type, typeToMatcher(type));
+  noticeErrorStringTest(type, typeToMatcher(type));
+  noticeErrorWithParamsTest(type, typeToMatcher(type));
+  multipleMatchingErrorsTest(type, typeToMatcher(type));
+})
+
+function noticeErrorTest(type, supportRegOrESMWorker) {
+  testDriver.test(`${type} - noticeError generates and sends an error object`, supportRegOrESMWorker, function (t, browser, router) {
+    let assetURL = router.assetURL(`worker/${type}-worker.html`, {
+      init,
+      workerCommands: [
+        () => newrelic.noticeError(new Error('test'))
+      ].map(x => x.toString())
+    })
+
+    let loadPromise = browser.get(assetURL)
+    let errPromise = router.expectErrors()
+
+    Promise.all([errPromise, loadPromise]).then(([errResponse]) => {
+      const { err } = JSON.parse(errResponse.body)
+      checkBasics(t, err)
+      t.deepEqual(err[0].custom, {worker: true}, 'Should not have correct custom attributes')
+      t.end()
+    }).catch(fail(t));
+	})
+}
+
+function noticeErrorStringTest(type, supportRegOrESMWorker) {
+  testDriver.test(`${type} - noticeError works with string argument`, supportRegOrESMWorker, function (t, browser, router) {
+    let assetURL = router.assetURL(`worker/${type}-worker.html`, {
+      init,
+      workerCommands: [`newrelic.noticeError("test")`]
+    })
+
+    let loadPromise = browser.get(assetURL)
+    let errPromise = router.expectErrors()
+
+    Promise.all([errPromise, loadPromise]).then(([errResponse]) => {
+      const { err } = JSON.parse(errResponse.body)
+      checkBasics(t, err)
+      t.deepEqual(err[0].custom, {worker: true}, 'Should not have correct custom attributes')
+      t.end()
+    }).catch(fail(t));
+	})
+}
+
+function noticeErrorWithParamsTest(type, supportRegOrESMWorker) {
+  testDriver.test(`${type} - noticeError generates and sends an error object with custom params`, supportRegOrESMWorker, function (t, browser, router) {
+    let assetURL = router.assetURL(`worker/${type}-worker.html`, {
+      init,
+      workerCommands: [
+        () => newrelic.noticeError(new Error('test'), { hi: 'mom' })
+      ].map(x => x.toString())
+    })
+
+    let loadPromise = browser.get(assetURL)
+    let errPromise = router.expectErrors()
+
+    Promise.all([errPromise, loadPromise]).then(([errResponse]) => {
+      const { err } = JSON.parse(errResponse.body)
+      checkBasics(t, err)
+      t.deepEqual(err[0].custom, { hi: 'mom', worker: true }, 'Should have correct custom attributes')
+      t.end()
+    }).catch(fail(t));
+	})
+}
+
+
+function multipleMatchingErrorsTest(type, supportRegOrESMWorker) {
```

Comment:
Do need a test on not _exactly_ matching errors **don't** aggregate? Though that may tie into the existing bug of aggregation disregarding column number. Could you just note a to-do to come back and add that once that bucketing is fixed?